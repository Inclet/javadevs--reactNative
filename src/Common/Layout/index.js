import React from "react";
import { View, StyleSheet } from "react-native";
import theme from "../../Styles/theme.style";
import common from "../../Styles/common.style";
import Icon from "react-native-vector-icons/Ionicons";
const Layout = ({ children, target, currentPage }) => {
  return (
    <View style={common.normalPageLayout}>
      {children}
      <View style={styles.bottomPaginator}>
        <Icon
          data-navigator="welcomeNavigator"
          style={[
            styles.paginatorStyle,
            currentPage === 1
              ? { color: theme.PRIMARY_COLOR }
              : { color: theme.LOW_CONTRAST_COLOR }
          ]}
          name="ios-radio-button-on"
          size={10}
          onPress={() => target.navigate("Welcome")}
        />
        <Icon
          data-navigator="aboutUsNavigator"
          data-testID="bottomNavigator"
          style={[
            styles.paginatorStyle,
            currentPage === 2
              ? { color: theme.PRIMARY_COLOR }
              : { color: theme.LOW_CONTRAST_COLOR }
          ]}
          name="ios-radio-button-on"
          size={10}
          onPress={() => target.navigate("AboutUs")}
        />
        <Icon
          data-navigator="authScreenNavigator"
          style={[
            styles.paginatorStyle,
            currentPage === 3
              ? { color: theme.PRIMARY_COLOR }
              : { color: theme.LOW_CONTRAST_COLOR }
          ]}
          name="ios-radio-button-on"
          size={10}
          onPress={() => target.navigate("AuthScreen")}
        />
      </View>
    </View>
  );
};

export default Layout;

export const styles = StyleSheet.create({
  bottomPaginator: {
    flex: 1,
    flexDirection: "row",
    alignItems: "flex-end",
    justifyContent: "center",
    marginBottom: 40
  },
  paginatorStyle: {
    padding: 5
  }
});
