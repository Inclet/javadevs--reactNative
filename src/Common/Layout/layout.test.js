import React from "react";
import { shallow } from "enzyme";
import Layout from "./index";

describe("Layout Component", () => {
  describe("RENDERING", () => {
    let wrapper,
      props,
      i = 0;
    beforeEach(() => {
      props = {
        target: {
          navigate: jest.fn()
        },
        currentPage: i++
      };
      wrapper = shallow(<Layout {...props} />);
    });
    it("Should match Snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("Should Navigate to welcome screen when clicking on the first paginator button", () => {
      wrapper.find('[data-navigator="welcomeNavigator"]').simulate("press");
      expect(props.target.navigate).toHaveBeenCalledWith("Welcome");
    });

    it("Should Navigate to About us screen when clicking on the second paginator button", () => {
      wrapper.find('[data-navigator="aboutUsNavigator"]').simulate("press");
      expect(props.target.navigate).toHaveBeenCalledWith("AboutUs");
    });

    it("Should Navigate to Auth screen when clicking on the third paginator button", () => {
      wrapper.find('[data-navigator="authScreenNavigator"]').simulate("press");
      expect(props.target.navigate).toHaveBeenCalledWith("AuthScreen");
    });
  });
});
