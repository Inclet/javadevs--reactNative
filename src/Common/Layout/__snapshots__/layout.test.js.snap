// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Should match Snapshot 1`] = `ShallowWrapper {}`;
